### PR TITLE
JENA-2202: Skip if longitude not available

### PR DESCRIPTION
**Pull request by afs**
Basic fix.

### CODE REVIEW

**Comment by neumarcx** [comment]
or set to zero if null it might be intentional.

**Review by kinow — APPROVED** [review]

**Comment by afs** [comment]
> or set to zero if null it might be intentional.

Null means "not present".

See report.

**Comment by neumarcx** [comment]
> > or set to zero if null it might be intentional.
> 
> Null means "not present".
> 
> See report.

reasonable

**Comment by rvesse** [comment]
> or set to zero if null it might be intentional.

Not sure we want [Null Island](https://en.wikipedia.org/wiki/Null_Island) being introduced here

File: jena-geosparql/src/main/java/org/apache/jena/geosparql/spatial/SpatialIndex.java [hunk 1]
```diff
@@ -457,7 +457,11 @@ public static final Dataset wrapModel(Model model) throws SpatialIndexException
             Resource feature = resIt.nextResource();
 
             Literal lat = feature.getRequiredProperty(SpatialExtension.GEO_LAT_PROP).getLiteral();
-            Literal lon = feature.getRequiredProperty(SpatialExtension.GEO_LON_PROP).getLiteral();
+            Literal lon = feature.getProperty(SpatialExtension.GEO_LON_PROP).getLiteral();
```
Comment (rvesse) [inline jena-geosparql/src/main/java/org/apache/jena/geosparql/spatial/SpatialIndex.java:460]: I notice there's no `null` check for `lat` either

Comment (afs) [reply jena-geosparql/src/main/java/org/apache/jena/geosparql/spatial/SpatialIndex.java:460]: The loop is over latitude properties:
```
ResIterator resIt = model.listResourcesWithProperty(SpatialExtension.GEO_LAT_PROP);
```

Comment (rvesse) [reply jena-geosparql/src/main/java/org/apache/jena/geosparql/spatial/SpatialIndex.java:460]: Ah yes, so this is guaranteed to be non-null or we wouldn't be in the loop to start with

**Pull request merged**